frontend/src/Calendar.js

```diff
@@ -2,19 +2,109 @@ import React from 'react'
 import FullCalendar from '@fullcalendar/react'
 import dayGridPlugin from '@fullcalendar/daygrid'
 
+import moment from 'moment'
+import timeGridPlugin from '@fullcalendar/timegrid'
+
+import {fetchUserCalendar} from './script'
+
 export default class Calendar extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      fetchDone: false,
+      schedulesTimes: [],
+      scheduleIdx: 0                 // Current schedule
+    }
+  }
+
+  componentDidMount(){
+    this.getAllSchedules()
+  }
+
+  async getAllSchedules() {
+    const schedules = await fetchUserCalendar()
+    console.log(schedules)
+    for(const schedule of schedules) {
+      this.getMeetingTimes(schedule)
+    }
+    
+    this.setState({
+      fetchDone: true
+    })
+    console.log(this.state.schedulesTimes)
+  }
+
+  getNextSchedule() {
+    if(this.state.scheduleIdx < this.state.schedulesTimes.length - 1) {
+      this.setState({scheduleIdx : this.state.scheduleIdx + 1})
+    }
+  }
+
+  getPrevSchedule() {
+    if(this.state.scheduleIdx > 0) {
+      this.setState({scheduleIdx : this.state.scheduleIdx - 1})
+    }
+  }
+
+  getMeetingTimes(schedule) {
+    let combinedMeetingTimes = []
+    console.log(schedule)
+    for (const course of schedule.courses) {
+      // Lecture Sections
+      const lectureTime = this.getScheduleSections(course.name, course.lectureSection.meetingTimes)
+      // Lab Sections
+      if (course.labSection) {
+        const labTime = this.getScheduleSections(course.name, course.labSection.meetingTimes)
+        combinedMeetingTimes = combinedMeetingTimes.concat(labTime)
+      }
+      combinedMeetingTimes = combinedMeetingTimes.concat(lectureTime)
+    }
+    //console.log(combinedMeetingTimes)
+    this.setState({
+      schedulesTimes:  [...this.state.schedulesTimes, combinedMeetingTimes]
+    })
+    //return combinedMeetingTimes
+  }
+  
+  getScheduleSections(courseName, meetingTimes) {
+    const lastSunday = this.getLastMonday(Date.now())
```

Comment:
The variable and function names here are inconsistent - I'm guessing one of them should be changed to Sunday/Monday?

Reply:
Thanks for pointing this out. I changed it to Sunday, since this is technically the start of the week, rather than Monday.